Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1009,5 +1028,55 @@ public void When_asserting_a_document_has_a_null_element_it_should_fail()
         }
 
         #endregion
+
+        #region xsd
+
+        [Fact]
+        public void Xsd_validation_for_valid_xml_should_work()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<PurchaseOrder xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'>
+                    <ShipTo>
+                        <name>Shepard</name>
+                        <street>Street</street>
+                    </ShipTo>
+                    <BillTo>
+                        <name>Billy</name>
+                        <street>Street</street>
+                    </BillTo>
+                </PurchaseOrder>");
+
+            // Act
+            Action act = () => document.Should().BeValidByXsd(XsdSample);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void Xsd_validation_for_invalid_xml_should_throw()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<PurchaseOrder xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'>
+                    <BillTo>
+                        <name>Billy</name>
+                        <street>Street</street>
+                    </BillTo>
+                    <ShipTo>
+                        <name>Shepard</name>
+                        <street>Street</street>
+                    </ShipTo>
+                </PurchaseOrder>");
+
+            // Act
+            Action act = () => document.Should().BeValidByXsd(XsdSample);
+
+            // Assert
+            act.Should().Throw<XunitException>();
```

Comment:
Please add `WithMessage` check to validate correctly build failure message sentence including the XSD validation error message and `because` string.